test-utils/generators/ApplicationRequestGenerator.scala

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package generators
+
+import models._
+import org.scalacheck._
+import wolfendale.scalacheck.regexp.RegexpGen
+
+trait ApplicationRequestGenerator extends Generators {
+  implicit lazy val arbitraryEoriNumberGen: Arbitrary[EoriNumber] = Arbitrary(
+    RegexpGen.from("^[A-Z]{2}[0-9A-Z]{12}$").map(EoriNumber)
+  )
+
+  implicit lazy val arbitraryUploadedDocument: Arbitrary[UploadedDocument] = Arbitrary {
+    for {
+      id       <- alphaStringsWithMaxLength(36)
+      fileName <- stringsWithMaxLength(100)
+      url      <- stringsWithMaxLength(100)
+      public   <- Gen.oneOf(true, false)
+      mimeType <- Gen.oneOf(
+                    "application/pdf",
+                    "application/msword",
+                    "application/vnd.ms-excel",
+                    "image/jpeg",
+                    "image/png",
+                    "text/plain"
+                  )
+      size     <- Gen.choose(1L, 1000000000000L)
+    } yield UploadedDocument(id, fileName, url, public, mimeType, size)
+  }
+
+  implicit lazy val arbitraryContactDetails: Arbitrary[ContactDetails] = Arbitrary {
+    for {
+      contactName      <- stringsWithMaxLength(70)
+      contactEmail     <- stringsWithMaxLength(70)
+      contactTelephone <- Gen.option(stringsWithMaxLength(9))
+    } yield ContactDetails(contactName, contactEmail, contactTelephone)
+  }
+
+  implicit lazy val arbitraryEoriDetails: Arbitrary[EORIDetails] = Arbitrary {
+    for {
+      eori         <- arbitraryEoriNumberGen.arbitrary
+      businessName <- stringsWithMaxLength(100)
+      addressLine1 <- stringsWithMaxLength(100)
+      addressLine2 <- stringsWithMaxLength(100)
+      addressLine3 <- stringsWithMaxLength(100)
+      postCode     <- stringsWithMaxLength(10)
+      countryCode  <- Gen.oneOf("UK", "JP", "FR", "DE", "IT", "ES", "US")
+    } yield EORIDetails(
+      eori.value,
+      businessName,
+      addressLine1,
+      addressLine2,
+      addressLine3,
```

Comment:
I think the EORIDetails contains a city field. Will `addressLine3` be used to hold the city?